packages/sources/tp/src/transport/price.ts

```diff
@@ -32,8 +33,28 @@ const isNum = (i: number | undefined) => typeof i === 'number'
 
 let providerDataStreamEstablishedUnixMs: number
 
+/*
+TP and ICAP EAs currently do not receive asset prices during off-market hours. When a heartbeat message is received during these hours,
+we update the TTL of cache entries that EA is requested to provide a price during off-market hours.
+ */
+const updateTTL = async (transport: WebSocketTransport<WsTransportTypes>) => {
+  // Get current active entries in the subscription set
+  const sSet = await transport.subscriptionSet.getAll()
+  // For each entry in sSet, calculate the cache key and try to update the ttl of an entry
+  sSet.forEach((param) => {
+    const key = calculateCacheKey({
+      transportName: transport.name,
+      data: param,
+      adapterName: transport.responseCache.adapterName,
+      endpointName: transport.responseCache.endpointName,
+      adapterSettings: transport.responseCache.adapterSettings,
+    })
+    transport.responseCache.cache.setTTL(key, transport.responseCache.adapterSettings.CACHE_MAX_AGE)
```

Comment:
is there any concern if `CACHE_MAX_AGE` is < heartbeat frequency? Might be worth documenting in the README as well 🤔 .

Reply:
If the default `CACHE_MAX_AGE` (90 seconds) is changed and is lower than heartbeat frequency (60 seconds), then obviously  there will be nothing to update and this feature won't work. Likewise if heartbeat frequency is changed the same will happen. Agree that we can probably inform somehow this in readme so that `CACHE_MAX_AGE` should not be less than 60 sec.

Reply:
Or  better if we error when starting the EA, wdyt ? @mmcallister-cll